src/data/multiset.lean

```diff
@@ -501,6 +501,28 @@ theorem range_subset {m n : ℕ} : range m ⊆ range n ↔ m ≤ n := range_subs
 
 @[simp] theorem not_mem_range_self {n : ℕ} : n ∉ range n := not_mem_range_self
 
+/- Ico -/
+
+/-- `Ico n m` is the multiset lifted from the list `Ico n m`,
+  that is, the set `{n, n+1, ..., m-1}`. -/
+def Ico (n m : ℕ) : multiset ℕ := Ico n m
+
+namespace Ico
+
+@[simp] theorem mem {n m l : ℕ} : l ∈ Ico n m ↔ n ≤ l ∧ l < m := list.Ico.mem
+
+@[simp] theorem Ico_range (n : ℕ) : Ico 0 n = range n :=
+by dsimp [Ico]; congr; dsimp [list.Ico]; rw list.range_eq_range'
+
+@[simp] theorem empty (n : ℕ) : Ico n n = ∅ :=
+by dsimp [Ico]; congr; simp
+
+@[simp] theorem succ_singleton (n : ℕ) : Ico n (n+1) = singleton n :=
+by dsimp [Ico]; congr; simp
+
+lemma pred_singleton {m : ℕ} (h : m > 0) : Ico (m-1) m = {m-1} :=
```

Comment:
why do you use a different singleton representation here? `singleton` vs `{..}`
Do they behave differently?

Reply:
Oops, I think I just wrote them at different times. I'll make both `{..}`, but I have no reason at the moment to prefer one over the other.

Reply:
I guess using `{..}` makes the statements look more like the corresponding statements for `list`, which is nice.